### PR TITLE
Fix the issue that PinotFs doesn't init before the first get in SegmentGenerationJobRunner

### PR DESCRIPTION
**Pull request by fx19880617**
## Description
Fix the issue that PinotFs doesn't init before the first get in SegmentGenerationJobRunner

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)
## Release Notes
If you have tagged this as either backward-incompat or release-notes,
you MUST add text here that you would like to see appear in release notes of the
next release.

If you have a series of commits adding or enabling a feature, then
add this section only in final commit that marks the feature completed.
Refer to earlier release notes to see examples of text

## Documentation
If you have introduced a new feature or configuration, please add it to the documentation as well.
See https://docs.pinot.apache.org/developers/developers-and-contributors/update-document

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
I think PinotFS should always be initialized first before launching the job. Is there a scenario where we don't get a chance to init the PinotFS?

File: pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/test/java/org/apache/pinot/plugin/stream/kafka20/KafkaPartitionLevelConsumerTest.java [hunk 1]
```diff
@@ -54,11 +54,11 @@
   private static final String TEST_TOPIC_2 = "bar";
   private static final int NUM_MSG_PRODUCED_PER_PARTITION = 1000;
 
-  private static MiniKafkaCluster kafkaCluster;
-  private static String brokerAddress;
+  private MiniKafkaCluster kafkaCluster;
```
Comment (mayankshriv) [inline pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/test/java/org/apache/pinot/plugin/stream/kafka20/KafkaPartitionLevelConsumerTest.java:57]: Curious on how the changes in these files are related to the fix?

Comment (fx19880617) [reply pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/test/java/org/apache/pinot/plugin/stream/kafka20/KafkaPartitionLevelConsumerTest.java:57]: I was seeing everytime the tests failed on this file, so just modify it to play around

**Review by Jackie-Jiang — APPROVED** [review]
LGTM.
Let's address the kafka test issue in a separate PR?

**Pull request merged**